Makefile

```diff
@@ -4,29 +4,46 @@ EXE := concurrent.exe state.exe ref.exe transaction.exe echo.exe delimcc.exe \
 	eratosthenes.exe pipes.exe loop.exe clone_is_tricky.exe fringe.exe \
 	algorithmic_differentiation.exe
 
+MVAR_EXE := chameneos_systhr.exe chameneos_lwt.exe chameneos_monad.exe chameneos-ghc.exe chameneos.exe MVar_test.exe
+
 all: $(EXE)
+mvar_all : $(MVAR_EXE)
 
 concurrent.exe: sched.mli sched.ml concurrent.ml
-	ocamlopt -o concurrent.exe sched.mli sched.ml concurrent.ml
+	@dune build concurrent.exe
 
 echo.exe: aio/aio.mli aio/aio.ml aio/echo.ml
-	$(MAKE) -C aio
-	cp aio/echo.native echo.exe
+	dune build aio/echo.exe
+	cp _build/default/aio/echo.exe .
```

Comment:
None of this `cp` commands is necessary as the executables can be run as `dune run aio/echo.exe` for example.